pkg/transportrequest/solman/create.go

```diff
@@ -0,0 +1,112 @@
+package solman
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/config/validation"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+	"io"
+	"strings"
+)
+
+// CreateAction Collects all the properties we need for creating a transport request
+type CreateAction struct {
+	Connection          Connection
+	ChangeDocumentID    string
+	DevelopmentSystemID string
+	CMOpts              []string
+}
+
+// Create collects everything which is needed for creating a transport request
+type Create interface {
+	WithConnection(Connection)
+	WithChangeDocumentID(string)
+	WithDevelopmentSystemID(string)
+	WithCMOpts([]string)
+	Perform(command Exec) (string, error)
+}
+
+// WithConnection specifies all the connection details which
+// are required in order to connect so SOLMAN
+func (a *CreateAction) WithConnection(c Connection) {
+	a.Connection = c
+}
+
+// WithChangeDocumentID specifies the change document for that
+// the transport request is created.
+func (a *CreateAction) WithChangeDocumentID(id string) {
+	a.ChangeDocumentID = id
+}
+
+// WithDevelopmentSystemID specifies the development system ID.
+func (a *CreateAction) WithDevelopmentSystemID(id string) {
+	a.DevelopmentSystemID = id
+}
+
+// WithCMOpts sets additional options for calling the
+// cm client tool. E.g. -D options. Useful for troubleshooting
+func (a *CreateAction) WithCMOpts(opts []string) {
+	a.CMOpts = opts
+}
+
+// Perform creates a new transport request
+func (a *CreateAction) Perform(command Exec) (string, error) {
+
+	log.Entry().Infof("Creating new transport request via '%s'.", a.Connection.Endpoint)
+
+	missingParameters, err := validation.FindEmptyStringsInConfigStruct(*a)
+
+	if err == nil {
+		notInitialized := len(missingParameters) != 0
+		if notInitialized {
+			err = fmt.Errorf("the following parameters are not available %s", missingParameters)
+		}
+	}
+
+	var transportRequestID string
+
+	if err == nil {
+		if len(a.CMOpts) > 0 {
+			command.SetEnv([]string{fmt.Sprintf("CMCLIENT_OPTS=%s", strings.Join(a.CMOpts, " "))})
+		}
+
+		oldStdout := command.GetStdout()
```

Comment:
TransportRequestId is written to `stdout`. I don't see any reason for redirecting `stderr` as well. That can stay like it is - it will be delivered to the logging framework.